projects/lights_control/src/main.c

```diff
@@ -3,19 +3,109 @@
 #include "can.h"
 #include "gpio.h"
 #include "interrupt.h"
+#include "log.h"
 #include "status.h"
 #include "wait.h"
 
+#include "lights_blinker.h"
+#include "lights_board_type.h"
+#include "lights_can.h"
+#include "lights_can_config.h"
+#include "lights_gpio.h"
+#include "lights_gpio_config.h"
+#include "lights_gpio_config_front.h"
+#include "lights_signal_fsm.h"
+#include "lights_strobe.h"
+
+#define LIGHTS_SIGNAL_BLINKER_DURATION 500
+#define LIGHTS_SIGNAL_SYNC_COUNT 5
+#define LIGHTS_STROBE_BLINKER_DURATION 500
+
+static LightsSignalFsm s_signal_fsm = { 0 };
+
+static LightsCanStorage s_lights_can_storage = { 0 };
+
+static LightsStrobeStorage s_lights_strobe = { 0 };
+
+static const GPIOAddress s_board_type_address = { .port = GPIO_PORT_B, .pin = 13 };
+
 int main(void) {
-  Event e;
+  CANSettings can_settings = {
+    .bitrate = CAN_HW_BITRATE_500KBPS,
+    .rx_event = LIGHTS_EVENT_CAN_RX,
+    // clang-format off
+    .tx = {
+      .port = GPIO_PORT_A,
+      .pin = 12,
+    },
+    .rx = {
+      .port = GPIO_PORT_A,
+      .pin = 11,
+    },
+    // clang-format on
+    .tx_event = LIGHTS_EVENT_CAN_TX,
+    .fault_event = LIGHTS_EVENT_CAN_FAULT,
+    .loopback = false,
+  };
 
-  // Initialize the various driver control devices
+  // Initialize the libraries.
   gpio_init();
   interrupt_init();
   soft_timer_init();
   event_queue_init();
 
+  // Getting board type.
+  GPIOState state = 0;
```

Comment:
Don't set enums to 0.

Reply:
Oh good point. I should do NUM_ENUMS right?